repos/system_upgrade/common/actors/peseventsscanner/libraries/peseventsscanner_repomap.py

```diff
@@ -0,0 +1,296 @@
+from leapp.libraries.common.config.version import get_source_major_version, get_target_major_version
+from leapp.libraries.stdlib import api
+
+
+DEFAULT_PESID = {
+    '7': 'rhel7-base',
+    '8': 'rhel8-BaseOS',
+    '9': 'rhel9-BaseOS'
+}
+
+
+def _get_channel_prio(pesid_repo):
+    priorities = {
+        'beta': 0,
+        'ga': 1,
+    }
+    return priorities.get(pesid_repo.channel, 10)
+
+
+class RepoMapDataHandler(object):
+    """
+    Provide the basic functionality to work with the repository data easily.
+    """
+
+    def __init__(self, repo_map, default_channels=None, prio_channel=None):
+        """
+        Initialize the object based on the given RepositoriesMapping msg.
+
+        Expects that msg contains just stuff related for the current IPU
+        (at least mapping and repos for the used upgrade path and architecture).
```

Comment:
Maybe we can place an example of such message here?

Reply:
I am not sure it will make stuff better any way if we put an exmple here. It's expected to have always valid message for the working stuff. It means in case of upgrade 7 -> 8, we do not expect to do any checks whether such a mapping exists, because it has to exist if a valid repomap data is delivered and scanner actor takes care partially about that for us. otherwise we should put inside various other checks, which we do not want to do again. that's the point of the comment. so we know we have mapping for this particular IPU and architecture (and currently we do not want to raise duplicated errors when data is missing).